hw1/tasks/task05.py

```diff
@@ -0,0 +1,18 @@
+"""
+Given a list of integers numbers "nums".
+You need to find a sub-array with length less equal to "k", with maximal sum.
+The written function should return the sum of this sub-array.
+Examples:
+    nums = [1, 3, -1, -3, 5, 3, 6, 7], k = 3
+    result = 16
+"""
+from typing import List
+
+
+def find_maximal_subarray_sum(nums: List[int], k: int) -> int:
+    result = nums[0]
+    for i in range(1, k+1):
```

Comment:
what's the purpose of outer for-loop?
it seems that only last iteration will look for subarray with 'k' length

Reply:
It's needed, cause the task is "to find a sub-array with length less equal to "k"". For example, for array [1,1,-1,-1,-1,-1] and k = 4 the answer is [1,1].

Reply:
but for [1,1,-1,-1,-1,-1] and k = 4 the result should be 0. we need maximal SUM of subarrays with length == k
try to write manually first 2 iterations, you'll see that outer loop is not needed